pkg/handler/collector/collector.go

```diff
@@ -109,17 +113,43 @@ func Collect(ctx context.Context, emitter Emitter, handleErr ErrHandler) error {
 	return nil
 }
 
-// Publish is used by NATS JetStream to stream the documents and send them to the processor
+// Publish takes the "document" collected by the collectors and stores it into a blob store for
+// retrieval by the processor/ingestor. A CDEvent is created to transmit the key (which is the
+// sha256 of the collected "document"). This also fixes the issues where the "document" was too large
+// to be sent across the event stream.
 func Publish(ctx context.Context, d *processor.Document) error {
 	logger := logging.FromContext(ctx)
+	blobStore := blob.FromContext(ctx)
+
 	docByte, err := json.Marshal(d)
 	if err != nil {
 		return fmt.Errorf("failed marshal of document: %w", err)
 	}
-	err = emitter.Publish(ctx, emitter.SubjectNameDocCollected, docByte)
+
+	err = blobStore.Write(ctx, getHash(d.Blob), docByte)
+	if err != nil {
+		return fmt.Errorf("failed write document to blob store: %w", err)
+	}
+
+	cdEvent, err := events.CreateArtifactPubEvent(ctx, getHash(d.Blob))
+	if err != nil {
+		return fmt.Errorf("failed create an event: %w", err)
+	}
+
+	keyByte, err := json.Marshal(cdEvent)
+	if err != nil {
+		return fmt.Errorf("failed marshal of document key: %w", err)
+	}
+
+	err = emitter.Publish(ctx, emitter.SubjectNameDocCollected, keyByte)
```

Comment:
same here, don't ever do `err = `, always `err :=` unless explicitly setting an err that is outside an if or for statment.